src/lib.rs

```diff
@@ -1,334 +1,201 @@
+        for (module, file_name) in modules {
+            if cfg!(feature = "protobuf-codec") {
+                writeln!(f, "pub mod {};", module).unwrap();
+                continue;
+            }
 
-    pub fn rustfmt(file_path: &Path) {
-        let output = Command::new("rustfmt")
-            .arg(file_path.to_str().unwrap())
-            .output();
-        if !output.map(|o| o.status.success()).unwrap_or(false) {
-            eprintln!("Rustfmt failed");
+            let mut level = 0;
+            for part in module.split('.') {
+                writeln!(f, "pub mod {} {{", part).unwrap();
+                level += 1;
+            }
+            writeln!(f, "include!(\"{}.rs\");", file_name,).unwrap();
+            if Path::new(&format!("{}/wrapper_{}.rs", self.out_dir, file_name)).exists() {
+                writeln!(f, "include!(\"wrapper_{}.rs\");", file_name,).unwrap();
+            }
+            writeln!(f, "{}", "}\n".repeat(level)).unwrap();
         }
     }
 
-    pub fn generate_wrappers<T: AsRef<str>>(file_names: &[T], out_dir: &str, gen_opt: GenOpt) {
-        for file in file_names {
-            let gen = super::wrapper::WrapperGen::new(file.as_ref(), gen_opt);
-            gen.write(out_dir);
+    fn prep_out_dir(&self) {
+        if Path::new(&self.out_dir).exists() {
+            fs::remove_dir_all(&self.out_dir).unwrap();
         }
+        fs::create_dir_all(&self.out_dir).unwrap();
     }
 
-    /// Returns a list of module names corresponding to the Rust files in a directory.
-    ///
-    /// Note that this does not read the files so will miss inline modules, it only
-    /// looks at filenames,
-    pub fn module_names_for_dir(directory_name: &str) -> Vec<String> {
-        let mut mod_names: Vec<_> = fs::read_dir(directory_name)
+    // List all `.rs` files in `self.out_dir`.
+    fn list_rs_files(&self) -> impl Iterator<Item = PathBuf> {
+        fs::read_dir(&self.out_dir)
             .expect("Couldn't read directory")
             .filter_map(|e| {
-                let file_name = e.expect("Couldn't list file").file_name();
-                let file_name = file_name.to_string_lossy();
-                if !file_name.ends_with(".rs") {
-                    return None;
+                let path = e.expect("Couldn't list file").path();
+                if path.extension() == Some(std::ffi::OsStr::new("rs")) {
+                    Some(path)
+                } else {
+                    None
                 }
-                file_name
-                    .split(".rs")
-                    .next()
-                    .filter(|n| !n.starts_with("wrapper_"))
-                    .map(ToOwned::to_owned)
             })
-            .collect();
+    }
+}
 
-        mod_names.sort();
-        mod_names
+impl Default for Builder {
+    fn default() -> Builder {
+        Builder::new()
     }
 }
 
-#[cfg(feature = "prost-codec")]
-pub use prost_imps::*;
+bitflags! {
+    pub struct GenOpt: u32 {
```

Comment:
Why always expose them since rust-protobuf will simply ignore them?

Reply:
It seemed like a lot of `cfg`s for not much benefit, but I could at least cfg the setter.